### PR TITLE
First functional version

### PR DESCRIPTION
**Pull request by agomezl**
`AFA` data type and examples

### CODE REVIEW

**Comment by asr** [comment]
You can merge on this repo, so please do it.

File: src/Data/Automata/AFA.hs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+-- | Two-way deterministic finite automata
```
Comment (asr) [inline src/Data/Automata/AFA.hs:1]: Wrong documentation.

**Comment by agomezl** [comment]
I will fix some issues and add `lab1.html` on another pull request.

**Comment by asr** [comment]
OK.

**Pull request closed without merging**